### PR TITLE
Feature/civ 3388

### PR DESCRIPTION
**Pull request by MMNycz**
**Before creating a pull request make sure that:**

- [X] commit messages are meaningful and follow good commit message guidelines
- [X] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-3388

### Related PR ###
https://github.com/hmcts/civil-general-apps-ccd-definition/pull/265

### Change description ###

Update caseData with new fields for event details and additional information (events "Refer to judge" and "Refer to Legal Advisor")

Create handler for user events Refer to judge" and "Refer to Legal Advisor"

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/model/genapplication/GACaseLocation.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder(toBuilder = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class GACaseLocation {
```
Comment (karthick-mohanasundaram-HMCTS) [inline src/main/java/uk/gov/hmcts/reform/civil/model/genapplication/GACaseLocation.java:12]: @MMNycz is this part of your change?

**Review by karthick-mohanasundaram-HMCTS — APPROVED** [review]

**Comment by karthick-mohanasundaram-HMCTS** [comment]
image

**Comment by karthick-mohanasundaram-HMCTS** [comment]
image

**Comment by karthick-mohanasundaram-HMCTS** [comment]
image

**Comment by karthick-mohanasundaram-HMCTS** [comment]
image

**Comment by karthick-mohanasundaram-HMCTS** [comment]
Summary page
image

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -7,7 +7,7 @@ import uk.gov.hmcts.contino.AppPipelineDsl
 def type = "java"
 def product = "civil"
 def component = "general-applications"
-def camundaBranch = "master"
+def camundaBranch = "feature/CIV-3388"
```
Comment (smitatalawadekar-hmcts) [inline Jenkinsfile_CNP:10]: Will need to change it back to master when testing is done

**Review by Azam-Hmcts — APPROVED** [review]
A part from smit comment, every thing is fine to me.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ReferToJudgeOrLegalAdvisorHandler.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        CaseData caseData = callbackParams.getCaseData();
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .errors(courtAssignedValidation(callbackParams))
+            .build();
+    }
+
+    public List<String> courtAssignedValidation(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        YesOrNo courtAssigned = caseData.getIsCcmccLocation();
```
Comment (smitatalawadekar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ReferToJudgeOrLegalAdvisorHandler.java:55]: would suggest renaming this field ... If isCCMCCLocation is YES then the local court is not assigned. So the condition in next few lines will not be confusing (currently it looks like courtAssigned is NO then add error that court is assigned)

**Comment by karthick-mohanasundaram-HMCTS** [comment]
Smoke Test

image

**Comment by karthick-mohanasundaram-HMCTS** [comment]
API

image

**Comment by karthick-mohanasundaram-HMCTS** [comment]
Functional multiparty
image

**Comment by karthick-mohanasundaram-HMCTS** [comment]
Functional 1v1
image

**Review by vasudevganesanhmcts — APPROVED** [review]

**Pull request merged**